### PR TITLE
[CLAP] Support batched inputs for CLAP. Fixes pipeline issues

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Support batching the `is_longer` part of the input for `zero_shot_audio_classification`. The model was not able to run batched inputs. Added a test to support this.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for working on this!

**Comment by ArthurZucker** [comment]
Just need to update the expected values for the doctest and will merge

File: src/transformers/models/clap/feature_extraction_clap.py [hunk 1]
```diff
@@ -347,6 +347,9 @@ def __call__(
         if isinstance(input_mel[0], List):
             input_mel = [np.asarray(feature, dtype=np.float64) for feature in input_mel]
 
+        if isinstance(is_longer, List):
```
Comment (ydshieh) [inline src/transformers/models/clap/feature_extraction_clap.py:350]: I didn't look into the details, but `is_longer` is always a list, as it is declared as `is_longer = []` above, and it keeps being as a list. 

Should we remove the condition here, and always do `is_longer = [[longer] for longer in is_longer]`?
And if possible, make a comment.

File: src/transformers/pipelines/zero_shot_audio_classification.py [hunk 2]
```diff
@@ -131,8 +132,8 @@ def _forward(self, model_inputs):
         outputs = self.model(**text_inputs, **model_inputs)
 
         model_outputs = {
-            "candidate_label": candidate_labels,
-            "logits_per_audio": outputs.logits_per_audio,
+            "candidate_labels": candidate_labels,
+            "logits": outputs.logits_per_audio,
```
Comment (ydshieh) [inline src/transformers/pipelines/zero_shot_audio_classification.py:136]: not against this, but is this necessary to fix the issue we have?

Comment (ArthurZucker) [reply src/transformers/pipelines/zero_shot_audio_classification.py:137]: it is, afterward we `pop` both arguments and there was a typo here.

**Review by ydshieh — APPROVED** [review]

**Pull request merged**